sdk/trace/trace_test.go

```diff
@@ -363,6 +363,21 @@ func TestStartSpanWithParent(t *testing.T) {
 	}
 }
 
+// Test we get a successful span as a new root if a nil context is sent in, as opposed to a panic. 
+// See https://github.com/open-telemetry/opentelemetry-go/issues/3109
+func TestStartSpanWithNilContext(t *testing.T) {
+	alwaysSampleTp := NewTracerProvider()
+	sampledTr := alwaysSampleTp.Tracer("AlwaysSampled")
+
+	defer func() {
+        if r := recover(); r != nil {
+            t.Error("unexpected panic creating span with nil context")
+        }
+    }()
+
+	_, _ = sampledTr.Start(nil, "should-not-panic")
```

Comment:
```suggestion
	_, _ = sampledTr.Start(context.TODO(), "should-not-panic")
```

Reply:
I think this doesn't help us here as `TODO()` returns a non-nil context, so the test isn't doing what we expect - https://go.dev/play/p/i9aqgGPU7PV (compared to https://go.dev/play/p/SXEK1EX_GzQ)

I think we'll need to instead just disable lint on that particular line